### PR TITLE
Feature: Rake task for revert trading activity for a particular user

### PR DESCRIPTION
**Pull request by mnaichuk**

### CODE REVIEW

**Review by mod — CHANGES_REQUESTED** [review]
Revert must be in Member model

File: app/models/member.rb [hunk 1]
```diff
@@ -84,6 +84,12 @@ def legacy_balance_for(currency:, kind:)
     end
   end
 
+  def revert_trading_activity!(trades)
+    trades.each do |t|
```
Comment (ysv) [inline app/models/member.rb]: Jut for you to not forget )
`trades.each(&:revert_trade)`

File: app/models/member.rb [hunk 2]
```diff
@@ -148,11 +154,14 @@ def fetch_email(payload)
     end
 
     def search(field: nil, term: nil)
+      term = "%#{term}%"
```
Comment (ysv) [inline app/models/member.rb:155]: nice solution !

File: lib/tasks/revert.rake [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+# Usage
+# for revert all trading activity for particular user:
+# $> bundle exec rake revert:trading_activity["admin@barong.io"]
+namespace :revert do
+  desc "Revert user trade activity."
+  task :trading_activity, [:member_email] => [:environment] do |_, args|
+
+    member = Member.find_by(email: args[:member_email])
+
+    # For each trade create revert Liabilities, Revenues and update User balances
```
Comment (ysv) [inline lib/tasks/revert.rake:13]: as an option you may pass trade ids through environment (same as member_email)

File: spec/lib/tasks/revert_spec.rb [hunk 4]
```diff
@@ -0,0 +1,119 @@
+# frozen_string_literal: true
+
+Rails.application.load_tasks
```
Comment (ysv) [inline spec/lib/tasks/revert_spec.rb:3]: It's better you test either Trade #revert_trade or Member #revert_trades!

**Review by ysv — APPROVED** [review]

**Review by mod — APPROVED** [review]

**Pull request merged**